packages/custom-webpack/src/custom-webpack-builder.spec.ts

```diff
@@ -50,45 +50,48 @@ const customWebpackFunctionObj = {
 	}
 };
 
-function createConfigFile(fileName: string) {
-	jest.mock(`${__dirname}/${fileName}`, () => customWebpackConfig, {virtual: true});
-}
-
-function createConfigFunction(fileName: string) {
-	jest.mock(`${__dirname}/${fileName}`, () => customWebpackFunction, {virtual: true})
+function createConfigFile(fileName: string, content: any) {
+	jest.unmock(`${__dirname}/${fileName}`);
+	jest.mock(`${__dirname}/${fileName}`, () => content, {virtual: true});
 }
 
 describe('CustomWebpackBuilder test', () => {
 	it('Should load webpack.config.js if no path specified', () => {
-		createConfigFile(defaultWebpackConfigPath);
+		createConfigFile(defaultWebpackConfigPath, customWebpackConfig);
 		CustomWebpackBuilder.buildWebpackConfig(__dirname as Path, {}, baseWebpackConfig, {});
 		expect(WebpackConfigMerger.merge).toHaveBeenCalledWith(baseWebpackConfig, customWebpackConfig, undefined, undefined);
 	});
 
 	it('Should load the file specified in configuration', () => {
 		const fileName = 'extra-webpack.config.js';
-		createConfigFile(fileName);
+		createConfigFile(fileName, customWebpackConfig);
 		CustomWebpackBuilder.buildWebpackConfig(__dirname as Path, {path: fileName}, baseWebpackConfig, {});
 		expect(WebpackConfigMerger.merge).toHaveBeenCalledWith(baseWebpackConfig, customWebpackConfig, undefined, undefined);
 	});
 
 	it('Should pass on merge strategies', () => {
-		createConfigFile(defaultWebpackConfigPath);
+		createConfigFile(defaultWebpackConfigPath, customWebpackConfig);
 		const mergeStrategies: MergeStrategies = {'blah': 'prepend'};
 		CustomWebpackBuilder.buildWebpackConfig(__dirname as Path, {mergeStrategies}, baseWebpackConfig, {});
 		expect(WebpackConfigMerger.merge).toHaveBeenCalledWith(baseWebpackConfig, customWebpackConfig, mergeStrategies, undefined);
 	});
 
 	it('Should pass on replaceDuplicatePlugins flag', () => {
-		createConfigFile(defaultWebpackConfigPath);
+		createConfigFile(defaultWebpackConfigPath, customWebpackConfig);
 		CustomWebpackBuilder.buildWebpackConfig(__dirname as Path, {replaceDuplicatePlugins: true}, baseWebpackConfig, {});
 		expect(WebpackConfigMerger.merge).toHaveBeenCalledWith(baseWebpackConfig, customWebpackConfig, undefined, true);
 	});
 
 	it('Should execute custom function on configuration', () => {
-		const fileName = 'custom-webpack.config.js';
-		createConfigFunction(fileName);
-		const mergedConfig = CustomWebpackBuilder.buildWebpackConfig(__dirname as Path, {path: fileName}, baseWebpackConfig, {env: 'prod'});
-		 expect(mergedConfig).toEqual(customWebpackFunctionObj);
+		const spy = jest.fn((config, options) => config);
+		createConfigFile(defaultWebpackConfigPath, spy);
+		CustomWebpackBuilder.buildWebpackConfig(__dirname as Path, {}, baseWebpackConfig, {env: 'prod'});
+		expect(spy).toHaveBeenCalledWith(baseWebpackConfig, {env: 'prod'});
+	});
+
+	it('Should apply custom function on configuration', () => {
```

Comment:
This test will fail if the second parameter won't be passed (while it checks whether the function was applied or not). So the applied function should ignore the second parameter (this way the test is truly isolated)

Reply:
Sorry I don't understand what you mean.

Reply:
Try to break the logic you've implemented. In `packages/custom-webpack/src/custom-webpack-builder.ts` at line 17 don't pass the build options.  
You'll see that it will fail both tests, although one of them checks that the function has been applied. So the function is still being but the test that verifies that fails for completely different reason (the second parameter is not passed).  
When writing unit test you (ideally) want to be able to identify where is the problem by just looking at the name of the failing test.  
Is that clearer now?

Reply:
I don't see the point on having such level of isolation for something that should work anyway (I would spend my entire life writing unit test if I had to apply this logic at work with our huge codebase).

It's done anyway.

Reply:
Thanks